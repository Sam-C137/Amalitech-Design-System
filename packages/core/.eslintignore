dist
loader
www
**/*.e2e.ts
**/*.spec.tsx
**/*.spec.ts
