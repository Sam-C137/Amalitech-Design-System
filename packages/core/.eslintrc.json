{
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "extends": [
    "plugin:@stencil-community/recommended"
  ],
  "ignorePatterns": [
    "**/*.stories.tsx",
    "**/*.spec.*",
    "**/*.e2e.*"
  ],
  "rules": {
    "react/jsx-no-bind": "off"
  }
}
