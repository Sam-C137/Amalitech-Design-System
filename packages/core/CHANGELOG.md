# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.2...v0.8.3) (2024-07-18)


### Bug Fixes

* min max value attribute changes ([8276d9d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/8276d9dd0d176071116497f01c12c92caecd5706))





## [0.8.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.1...v0.8.2) (2024-06-02)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.8.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.0...v0.8.1) (2024-06-01)


### Bug Fixes

* tests ([05989c4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/05989c4eaa674a902264d87c3044b6eb91794698))





# [0.8.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.7.0...v0.8.0) (2024-05-27)


### Features

* add double border to radio ([21d478d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/21d478d51df6c38aa88b8a1b429b76baa7371040))





# [0.7.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.2...v0.7.0) (2024-05-14)


### Features

* add clear method to select ([689be3d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/689be3df7d4c3f895ef8b2f11fc8bc24a2f78ab3))
* create select value accessor directive ([d5957c8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/d5957c85d858f02ec8506671ae7ac142421a34c5))





## [0.6.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.1...v0.6.2) (2024-05-06)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.6.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.0...v0.6.1) (2024-05-06)


### Bug Fixes

* removed default switch margin ([503f85d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/503f85da2f380b6b7a89e3a46015ea367d67a2ce))





# [0.6.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.5.0...v0.6.0) (2024-04-30)


### Bug Fixes

* pagination spacing ([00c99e4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/00c99e4e17edf1f4d145135d8f1da444f6cb6ed6))


### Features

* exposed totalPagesCount in pagination ([4105737](https://github.com/Amali-Tech/Amalitech-Design-System/commit/4105737d72ba70a4bc0b9a7fa67dca5e73c2bc5b))





# [0.5.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.4.0...v0.5.0) (2024-04-28)


### Features

* utilities align self ([ab106b6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/ab106b6f3752e20155c9b307d89cbcc4b12c79ec))





# [0.4.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.7...v0.4.0) (2024-04-28)


### Features

* add z-index to utilities ([75efd78](https://github.com/Amali-Tech/Amalitech-Design-System/commit/75efd78fbf0a22ce34139c7569a5ee894faaa088))





## [0.3.7](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.6...v0.3.7) (2024-04-27)


### Bug Fixes

* date picker open method: ([710e97b](https://github.com/Amali-Tech/Amalitech-Design-System/commit/710e97b29a3538231ada47ffddd7314be1766559))





## [0.3.6](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.5...v0.3.6) (2024-04-24)


### Bug Fixes

* centered the checkbox in it's label ([e7fbbaa](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e7fbbaaf3f79a6d0185777a0ce52a44a6b6f84e8))






## [0.3.5](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.4...v0.3.5) (2024-04-15)


### Bug Fixes

* build ([12140c9](https://github.com/Amali-Tech/Amalitech-Design-System/commit/12140c95e8be561d9f26d67c15b71c2beb50dee0))
* build ([a38dd2e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a38dd2e615bedc32ab7a68ce841a9b04e4f8267b))
* card tests ([5c99406](https://github.com/Amali-Tech/Amalitech-Design-System/commit/5c99406dc9796839ee06326fdaf2b8471e2d554d))
* starting with a value ([fb24ce3](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fb24ce3cf7510dc5cca9c868aa03b48e3aa8acdb))
* tests ([373434e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/373434e70de883671a1f828d1cbda137d717978a))
* tests ([eb11522](https://github.com/Amali-Tech/Amalitech-Design-System/commit/eb11522d23bf5b31c41228ba9f522998641c677f))
* tests ([e44f72d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e44f72d42a5b9fc587edfbeecafd3c1dc89b799a))
* tests ([799a98c](https://github.com/Amali-Tech/Amalitech-Design-System/commit/799a98cf03b6c008f749fdb754187abcca9cface))
* tests ([fcb7f47](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fcb7f47051af324d7d5ea1c0a285f5f2645788be))
* tests ([66ea8f8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/66ea8f888460610f550d88b51f1d4ec870444072))






## [0.3.4](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.3...v0.3.4) (2024-04-05)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.2...v0.3.3) (2024-04-04)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.1...v0.3.2) (2024-04-03)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.3.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.0...v0.3.1) (2024-04-03)


### Bug Fixes

* tests ([a1ca9e2](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a1ca9e26ce368e9d26b114605ad3869ccaa5dd8d))






# [0.3.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.2...v0.3.0) (2024-04-02)


### Features

* form submission for input & button ([2cc64d6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/2cc64d69378de5a3ec3e7c9da20f53e36e3a5dcc))






## [0.2.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.1...v0.2.2) (2024-03-28)

**Note:** Version bump only for package @amalitech-design-system/core
