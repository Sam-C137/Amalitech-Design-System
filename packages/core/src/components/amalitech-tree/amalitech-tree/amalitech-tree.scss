:host {
  display: block;
}

.tree-view {
  background-color: var(--layer, $light);
}
