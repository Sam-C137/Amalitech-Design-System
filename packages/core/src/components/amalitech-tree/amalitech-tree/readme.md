# amalitech-tree



<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description                          | Type     | Default                                                                                  |
| -------------- | --------------- | ------------------------------------ | -------- | ---------------------------------------------------------------------------------------- |
| `emptyState`   | `empty-state`   | To be shown when no node is rendered | `string` | ``{     "headline": "No items",     "description": "There are no items to display"   }`` |
| `selectedNode` | `selected-node` | selects a node at default            | `string` | `undefined`                                                                              |


## Methods

### `getSelectedNode() => Promise<string>`

Returns selected node

#### Returns

Type: `Promise<string>`



### `setFocus() => Promise<void>`

Sets focus on first menu item. Use this method instead of the global
`element.focus()`.

#### Returns

Type: `Promise<void>`



### `subscribeToSelect(cb: any) => Promise<void>`



#### Parameters

| Name | Type  | Description |
| ---- | ----- | ----------- |
| `cb` | `any` |             |

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
