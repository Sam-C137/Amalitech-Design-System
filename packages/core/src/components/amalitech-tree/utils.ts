export const getComponentIndex = (() => {
  let counter = 1;
  return function () {
    return `${counter++}`;
  };
})();