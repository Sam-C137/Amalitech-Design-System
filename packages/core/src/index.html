<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Stencil Component Starter</title>
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <style>
      h1 {
        text-align: center;
        font-style: normal;
        line-height: 2.5rem;
        font-family: var(--amalitech-font-variant-body-family);
        color: var(--amalitech-color-neutral-700);
        font-size: var(--amalitech-font-variant-heading-3-size);
        font-weight: var(--amalitech-font-weight-bold);
        margin: 0;
        padding: 0;
      }
    </style>
  </head>

  <body>
    <h1>This is ADS</h1>
  </body>
</html>
