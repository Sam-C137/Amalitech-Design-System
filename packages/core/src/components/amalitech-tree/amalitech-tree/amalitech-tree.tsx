import { Component, ComponentInterface, Element, h, Listen, Method, Prop, State, Watch } from '@stencil/core';

@Component({
  tag: 'amalitech-tree',
  styleUrl: 'amalitech-tree.scss',
  shadow: true,
})
export class AmalitechTree implements ComponentInterface {
  @Element() elm!: HTMLAmalitechTreeElement;

  /**
   * Checks for empty state
   */
  @State() empty: boolean = false;

  /**
   * To be shown when no node is rendered
   */
  @Prop() emptyState: string = `{
    "headline": "No items",
    "description": "There are no items to display"
  }`;

  /**
   * selects a node at default
   */
  @Prop({ mutable: true }) selectedNode: string;

  @State() internalEmptyState: any;

  @Watch('emptyState')
  parseEmptyState() {
    if (typeof this.emptyState === 'string') {
      this.internalEmptyState = JSON.parse(this.emptyState);
    } else {
      this.internalEmptyState = this.emptyState;
    }
  }

  @Listen('tree:tree-node-click')
  treeNodeClick(evt: any) {
    this.selectedNode = evt.detail.id;
    this.subscribers.forEach(cb => cb(evt.detail.value));
  }

  private subscribers: any[] = [];

  /**
   * Returns selected node
   */
  @Method()
  async getSelectedNode() {
    return this.selectedNode;
  }

  /**
   * 
   * @param cb
   */
  @Method()
  async subscribeToSelect(cb: any) {
    this.subscribers.push(cb);
  }

  @Listen('keydown', { target: 'window' })
  handleKeyDown(evt: KeyboardEvent) {
    const path = evt.composedPath();
    let menuItem = null;
    for (const elm of path) {
      // @ts-ignore
      if (elm.tagName === 'AMALITECH-TREE-NODE') {
        menuItem = elm;
      }
      if (elm !== this.elm) continue;
      if (evt.key === 'ArrowDown') {
        evt.preventDefault();
        this.focusNextItem(menuItem);
      } else if (evt.key === 'ArrowUp') {
        evt.preventDefault();
        this.focusPreviousItem(menuItem);
      }
    }
  }

  /**
   * Sets focus on first menu item. Use this method instead of the global
   * `element.focus()`.
   */
  @Method()
  async setFocus() {
    const firstMenuItem = this.getFirstItem();
    firstMenuItem?.setFocus();
  }

  private getFirstItem() {
    return this.elm.querySelector('amalitech-tree-node');
  }

  private focusNextItem(currentItem: any) {
    let nextItem: any = currentItem.nextElementSibling;
    do {
      if (nextItem && nextItem.tagName === 'AMALITECH-TREE-NODE' && !nextItem.disabled) {
        nextItem.setFocus();
        return;
      }
      if (!nextItem) {
        nextItem = this.elm.querySelector('amalitech-tree-node:first-child');
      } else {
        nextItem = nextItem.nextElementSibling;
      }
    } while (nextItem !== currentItem);
  }

  private focusPreviousItem(currentItem: any) {
    let previousItem: any = currentItem.previousElementSibling;
    do {
      if (previousItem && previousItem.tagName === 'AMALITECH-TREE-NODE' && !previousItem.disabled) {
        previousItem.setFocus();
        return;
      }
      if (!previousItem) {
        previousItem = this.elm.querySelector('amalitech-tree-node:last-child');
      } else {
        previousItem = previousItem.previousElementSibling;
      }
    } while (previousItem !== currentItem);
  }

  componentWillLoad() {
    this.parseEmptyState();
  }

  render() {
    if (this.empty) {
      return <div class="tree-view">{this.renderEmptyState()}</div>;
    } else
      return (
        <div class="tree-view">
          <slot></slot>
        </div>
      );
  }

  private renderEmptyState() {
    if (this.empty) return <p>Please add tree items</p>;
  }
}
