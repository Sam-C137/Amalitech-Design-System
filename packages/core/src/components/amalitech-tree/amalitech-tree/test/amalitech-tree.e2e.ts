import { newSpecPage } from '@stencil/core/testing';
import { AmalitechTree } from '../amalitech-tree';

describe('AmalitechTree', () => {
  it('renders empty state when empty prop is true', async () => {
    const page = await newSpecPage({
      components: [AmalitechTree],
      html: '<amalitech-tree empty></amalitech-tree>'
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders tree nodes when empty prop is false', async () => {
    const page = await newSpecPage({
      components: [AmalitechTree],
      html: '<amalitech-tree><amalitech-tree-node>Node 1</amalitech-tree-node></amalitech-tree>'
    });
    expect(page.root).toMatchSnapshot();
  });
});
