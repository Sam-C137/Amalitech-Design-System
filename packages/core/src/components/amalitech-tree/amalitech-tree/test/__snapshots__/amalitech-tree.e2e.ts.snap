// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmalitechTree renders empty state when empty prop is true 1`] = `
<amalitech-tree empty="">
  <mock:shadow-root>
    <div class="tree-view">
      <slot></slot>
    </div>
  </mock:shadow-root>
</amalitech-tree>
`;

exports[`AmalitechTree renders tree nodes when empty prop is false 1`] = `
<amalitech-tree>
  <mock:shadow-root>
    <div class="tree-view">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <amalitech-tree-node>
    Node 1
  </amalitech-tree-node>
</amalitech-tree>
`;
